flixel/system/scaleModes/ShaderScaleMode.hx

```diff
@@ -0,0 +1,102 @@
+package flixel.system.scaleModes;
+
+import flixel.FlxG;
+import flixel.util.typeLimit.OneOfTwo;
+
+#if !openfl_legacy
+import shaders.ScaleShaderFilter;
+import shaders.Nearest;
+import shaders.Bilinear;
+
+import openfl.filters.BitmapFilter;
+import openfl.display.Shader;
+import openfl.filters.ShaderFilter;
+import openfl.gl.GL;
+#end
+
+@:enum
+abstract ShaderScaleEnum(Int){
+	var NEAREST = 0;
+	var BILINEAR = 1;
+}
+
+class ShaderScaleMode extends flixel.system.scaleModes.BaseScaleMode
+{
+
+	#if !openfl_legacy
+	private var scaleX:Float;
+	private var scaleY:Float;
+	private var strength:Float;
+
+	public var filter:shaders.ScaleShaderFilter;
+	public var filters:Array<BitmapFilter>;
+
+	public function new(?shader:OneOfTwo<ShaderScaleEnum, openfl.display.Shader> = ShaderScaleEnum.NEAREST, ?scaleX:Float = 1, ?scaleY:Float = 1){
```

Comment:
There's no reason for the `?` here, the default values already make the params optional. Additionally, `?` turns basic types into nullable ones, creating unnecessary runtime overhead.

http://stackoverflow.com/questions/31914614/is-question-mark-operator-in-haxe-really-necessary/31916466#31916466